parcels/particle.py

```diff
@@ -146,6 +147,9 @@ def __init__(self):
         # Placeholder for explicit error handling
         self.exception = None
 
+    def __del__(self):
+        pass
+
```

Comment:
Proper coding style that guarantees avoiding memory leaks writes a `__del__` function for each and every class that is created. Destructors cannot be used indirectly from superclasses (e.g. `object`), because a destructor is individual to each class. That is the reason for one of the refactor-PRs (number currently not in my head).

Usually, one can call `super(<baseclass-here>, self).__del__()`. That is ideal - and usually should be the last line of any destructor anyway. Sadly and obviously, that doesn't work here, cause those base classes don't have a destructor themselves. If one cannot call the superclass-destructor, then the only thing one can call is `pass`.

Now, obviously the destructor should contain much more than just this `super()`-call (or a `pass`), when it comes to dynamically-allocated data. What that entails -> wikipedia or google on 'class destructors basics'.

Hope that helps.